equinox/nn/_composed.py

```diff
@@ -136,32 +139,58 @@ class Sequential(Module):
         Activation functions can be added by wrapping them in [`equinox.nn.Lambda`][].
     """
 
-    layers: tuple[_SequentialLayer, ...]
+    layers: tuple
 
-    def __init__(self, layers: Sequence[_SequentialLayer]):
+    def __init__(self, layers: Sequence[Module]):
         self.layers = tuple(layers)
 
+    @overload
     def __call__(self, x: Array, *, key: Optional[PRNGKeyArray] = None) -> Array:
+        ...
+
+    @overload
+    def __call__(
+        self, x: Array, state: State, *, key: Optional[PRNGKeyArray] = None
+    ) -> tuple[Array, State]:
+        ...
+
+    def __call__(
+        self,
+        x: Array,
+        state: State = sentinel,
+        *,
+        key: Optional[PRNGKeyArray] = None,
+    ) -> Union[Array, tuple[Array, State]]:
         """**Arguments:**
 
-        - `x`: Argument passed to the first member of the sequence.
-        - `key`: A `jax.random.PRNGKey`, which will be split and passed to every layer
-            to provide any desired randomness. (Optional. Keyword only argument.)
+        - `x`: passed to the first member of the sequence.
+        - `state`: If provided, then it is passed on to any layer which implements
+            `equinox._batch_norm._StatefulLayer`.
```

Comment:
"then it is passed to any layer which subclasses `eqx.nn.StatefulLayer`, which is used purely to indicate that this layer accepts and returns an extra `state` argument"